### PR TITLE
Fix control user issue

### LINKED ISSUE #15608: DisableIS is broken on pma 4.9.2 with controluser configured
### Describe the bug

There is a problem when `DisableIS` cfg option is true, and there is a `controluser` configured.

After logging in as root and clicking on a database name on the navigation tree (not to the "+", directly to the name), it pops up the error "An error has occurred while loading the navigation display". At that point, refreshing the page causes all databases except information_schema, mysql, and performance_schema to disappear.

I tried further investigating the issue, it is probably hiding a more serious bug related to the management of links for control user and logged in user. In particular, the problem started to appear from commit b8650115be7cc2d79d231723b7a6f1961e834d27 (which was buggy but it doesn't matter for this issue), the relevant change is the behaviour of instance member _current_user, which is no longer cached on error. This means that while before it was always returning ["",""], now it starts returning ["root","%"] when clicking on the navigation tree, which in turns triggers the bug.

I'm afraid there is something wrong with the whole `DatabaseInterface` class, it seems it is mixing up the control user link and the logged in user.

Please let me know if you need further information as I spent several hours troubleshooting this.

### Server configuration

- Operating system: Fedora 31
- Web server: Apache/2.4.41 (Fedora) OpenSSL/1.1.1d PHP/7.3.12
- Database version: 10.3.18-MariaDB-log - MariaDB Server
- PHP version: 7.3.12
- phpMyAdmin version: 4.9.2

### PR DESCRIPTION
**Pull request by MauricioFauth**
Partially reverts commit 2bc8afea5f47d0d409163754698ad13487410158.

Fixes #15608
Related to #14433

### CODE REVIEW

File: libraries/classes/DatabaseInterface.php [hunk 1]
```diff
@@ -2330,11 +2330,7 @@ public function getCurrentUserAndHost()
     {
         if (count($this->_current_user) == 0) {
             $user = $this->getCurrentUser();
-            if ($user === '@') {// Request did not succeed, please do not cache
-                return ['', ''];
```
Comment (williamdes) [inline libraries/classes/DatabaseInterface.php]: A lot of code relies on the access of the 2 indexes, nothing broken ?

**Review by williamdes — APPROVED** [review]

File: libraries/classes/DatabaseInterface.php (hunk 1 above)
Comment (mauriciofauth) [reply libraries/classes/DatabaseInterface.php]: Exploding `'@'` returns the same array: `explode('@', '@') == ['', '']`

**Comment by williamdes** [comment]
:rocket: thank you !

**Comment by williamdes** [comment]
Introduced #15724

**Pull request merged**